scripts/cleanupHusky.mjs

```diff
@@ -0,0 +1,72 @@
+/// @ts-check
+import * as childProcess from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+
+let changedHooksPath = false;
+
+const oldHuskyHooks = [
+  'applypatch-msg',
+  'commit-msg',
+  'post-applypatch',
+  'post-checkout',
+  'post-commit',
+  'post-merge',
+  'post-receive',
+  'post-rewrite',
+  'post-update',
+  'pre-applypatch',
+  'pre-auto-gc',
+  'pre-merge-commit',
+  'pre-push',
+  'pre-rebase',
+  'pre-receive',
+  'push-to-checkout',
+  'sendemail-validate',
+  'update',
+];
+
+//
+// Husky's postinstall script changes your local repo git config, so undo those changes.
+// Still respect the setting though if it's been set to something else.
+const hooksConfig = childProcess.spawnSync('git', ['config', 'core.hooksPath'], { encoding: 'utf-8' });
```

Comment:
for the sake of making this even faster you can first check if `husky` is a key in the `package.json`. It is much faster than invoking system binaries.

Reply:
when would that be true? This file only exists when there's no more husky in package.json...

Reply:
When you pull these changes for the first time since being merged into main:

 - you haven't ran this cleanup script yet
 - you will have husky in your git config
 - husky will no longer be in package.json because this PR removed it.

So if we made that change then it wouldn't do anything. Or am I missing something?

I'm not overly fussed about this. The previous husky postinstall shelled out exactly the same way, so this is a net no-change in perf (if mesurable at all)

Reply:
We expect to have this script run in postinstall for about the next 6 months. https://docs.google.com/document/d/18cT5QgQxxpZGJtvUj-Q0zOrnrQCkIQFL0XuZTdIq1Zc/edit#heading=h.rovirazb7wo1